src/telegram/error.py

```diff
@@ -208,21 +213,43 @@ class RetryAfter(TelegramError):
        :attr:`retry_after` is now an integer to comply with the Bot API.
 
     Args:
-        retry_after (:obj:`int`): Time in seconds, after which the bot can retry the request.
+        retry_after (:obj:`int` | :class:`datetime.timedelta`): Time in seconds, after which the
+            bot can retry the request.
+
+            .. versionchanged:: NEXT.VERSION
+                |time-period-input|
 
     Attributes:
-        retry_after (:obj:`int`): Time in seconds, after which the bot can retry the request.
+        retry_after (:obj:`int` | :class:`datetime.timedelta`): Time in seconds, after which the
+            bot can retry the request.
+
+            .. deprecated:: NEXT.VERSION
+                |time-period-int-deprecated|
 
     """
 
-    __slots__ = ("retry_after",)
+    __slots__ = ("_retry_after",)
+
+    def __init__(self, retry_after: TimePeriod):
+        self._retry_after: dtm.timedelta = parse_period_arg(  # type: ignore[assignment]
+            retry_after
+        )
+
+        if isinstance(self.retry_after, int):
```

Comment:
```suggestion
        if isinstance(retry_after, int):
```
feels weird to immediately call a property which unwraps the value just assigned to `_retry_after`. The pattern I would like is something like this:

```python
seconds_in_int = retry_after if isinstance(retry_after, int) else retry_after.total_seconds()
super().__init__("...")
```
this pattern avoids some repetition imo

Reply:
ah I didn't consider the behaviour of the message depending on the env var. The current way is fine then imo 👍🏽